src/NServiceBus.Core/DataBus/UseDataBusExtensions.cs

```diff
@@ -1,51 +1,78 @@
 ﻿namespace NServiceBus
 {
     using System;
+    using System.Collections.Generic;
     using DataBus;
 
     /// <summary>
     /// Extension methods to configure data bus.
     /// </summary>
-    public static class UseDataBusExtensions
+    public static partial class UseDataBusExtensions
     {
         /// <summary>
         /// Configures NServiceBus to use the given data bus definition.
         /// </summary>
         /// <param name="config">The <see cref="EndpointConfiguration" /> instance to apply the settings to.</param>
-        public static DataBusExtensions<T> UseDataBus<T>(this EndpointConfiguration config) where T : DataBusDefinition, new()
+        public static DataBusExtensions<TDataBus> UseDataBus<TDataBus, TDataBusSerializer>(this EndpointConfiguration config)
+            where TDataBus : DataBusDefinition, new()
+            where TDataBusSerializer : IDataBusSerializer, new()
         {
             Guard.AgainstNull(nameof(config), config);
-            var type = typeof(DataBusExtensions<>).MakeGenericType(typeof(T));
-            var extension = (DataBusExtensions<T>)Activator.CreateInstance(type, config.Settings);
-            var definition = (DataBusDefinition)Activator.CreateInstance(typeof(T));
-            config.Settings.Set("SelectedDataBus", definition);
 
-            config.EnableFeature<Features.DataBus>();
+            var dataBusSerializer = (IDataBusSerializer)Activator.CreateInstance(typeof(TDataBusSerializer));
```

Comment:
hmmm since the generic type is already stating `new()` constraint, can't we just create it without Activator? e.g.

```
var dataBusSerializer = IDataBusSerializer)new TDataBusSerializer();
```

Reply:
Agreed